bool arg
---
string path
uint32 size
